### PR TITLE
Make Wrapper task work again on Windows

### LINKED ISSUE #8078: wrapper task fails when the version 5.1 is specified.
### Current Behavior

wrapper task fail when the version 5.1(5.1-rc-1/5.1-rc-2) is specified.
\* When the version 5.0 is specified, an error does not occur.

```
.\gradlew.bat wrapper --gradle-version=5.1-rc-2
> Task :wrapper FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':wrapper'.
> Failed to write wrapper JAR to C:\Users\xxx\gradle\wrapper\gradle-wrapper.jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
1 actionable task: 1 executed
```

### Your Environment
WIndows 10

### PR DESCRIPTION
**Pull request by oehme**
Fixes #8078

### CODE REVIEW

File: subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java [hunk 1]
```diff
@@ -141,8 +141,8 @@ private void writeWrapperTo(File destination) {
         if (jarFileSource == null) {
             throw new GradleException("Cannot locate wrapper JAR resource.");
         }
-        try (InputStream stream = jarFileSource.openStream()) {
-            Files.copy(stream, destination.toPath(), REPLACE_EXISTING);
+        try (InputStream in = jarFileSource.openStream(); OutputStream out = new FileOutputStream(destination)) {
```
Comment (wolfs) [inline subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java]: I guess `Files.newOutputStream` has the same problem as `Files.copy`? Or can the problem be fixed by providing the right options to `Files.newOutputStream`?

Do you understand the differences between `Files` and when using `FileOutputStream`?

**Comment by oehme** [comment]
Files.copy tries to delete the target file first, which is what fails on Windows. An `OutputStream` on the other hand will overwrite its content without calling `delete`.

**Comment by oehme** [comment]
I just realized that this test will never do anything, since we are not actually using the wrapper we created. Using the wrapper and overwriting it at the same time is what is causing the issue.

**Comment by oehme** [comment]
`Files.newOutputStream` should also be fine, it doesn't try to delete the existing file, it only truncates it.

**Comment by wolfs** [comment]
@oehme Are you going to add a test where we actually use the wrapper to create a new wrapper script?

**Comment by oehme** [comment]
Yes I'm working on one.

**Pull request merged**